### PR TITLE
Update kubeadm troubleshooting guide

### PR DESCRIPTION
**Pull request by detiber**
Updates to the kubeadm troubleshooting guide

### CODE REVIEW

File: content/en/docs/setup/independent/troubleshooting-kubeadm.md [hunk 1]
```diff
@@ -87,122 +80,159 @@ sudo kubeadm reset
...
-Please note that this assumes `KUBELET_EXTRA_ARGS` hasn't already been set in the unit file.
+  ```sh
+  systemctl daemon-reload
+  systemctl restart kubelet
+  ```
 
-Then restart `kubelet`:
+#### Services with externalTrafficPolicy=Local are not reachable
+
+On nodes where the hostname for the kubelet is overriden uisng the `--hostname-override` option, kube-proxy will default to treating 127.0.0.1 as the node IP, which results in rejecting connections for Services configured for `externalTrafficPolicy=Local`. This situation can be verified by checking the output of `kubectl -n kube-system logs <kube-proxy pod name>`:
```
Comment (neolit123) [inline content/en/docs/setup/independent/troubleshooting-kubeadm.md]: `overriden uisng ` -> `overridden using`

File: content/en/docs/setup/independent/troubleshooting-kubeadm.md [hunk 2]
```diff
@@ -87,122 +80,159 @@ sudo kubeadm reset
...
 
-If you are using VirtualBox (directly or via Vagrant), you will need to
-ensure that `hostname -i` returns a routable IP address (i.e. one on the
-second network interface, not the first one). By default, it doesn't do this
-and kubelet ends-up using first non-loopback network interface, which is
-usually NATed. Workaround: Modify `/etc/hosts`, take a look at this
-`Vagrantfile`[ubuntu-vagrantfile](https://github.com/errordeveloper/k8s-playground/blob/22dd39dfc06111235620e6c4404a96ae146f26fd/Vagrantfile#L11) for how this can be achieved.
+- If you are using VirtualBox (directly or via Vagrant), you will need to
+  ensure that `hostname -i` returns a routable IP address. By default the first
+  interface is connected to a non-routeable host-only network. A work around
```
Comment (neolit123) [inline content/en/docs/setup/independent/troubleshooting-kubeadm.md]: `non-routeable` -> `non-routable` ?

File: content/en/docs/setup/independent/troubleshooting-kubeadm.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ This may be caused by a number of problems. The most common are:
   Check the system log file (e.g. `/var/log/message`) or examine the output from `journalctl -u kubelet`. If you see something like the following:
 
   ```shell
-  error: failed to run Kubelet: failed to create kubelet: 
+  error: failed to run Kubelet: failed to create kubelet:
   misconfiguration: kubelet cgroup driver: "systemd" is different from docker cgroup driver: "cgroupfs"
```
Comment (neolit123) [inline content/en/docs/setup/independent/troubleshooting-kubeadm.md]: `from docker` -> `from the Docker`

File: content/en/docs/setup/independent/troubleshooting-kubeadm.md [hunk 4]
```diff
@@ -27,12 +27,12 @@ If your cluster is in an error state, you may have trouble in the configuration
...
-[preflight] WARNING: ethtool not found in system path                           
+```sh
+[preflight] WARNING: ebtables not found in system path
+[preflight] WARNING: ethtool not found in system path
 ```
 
-Then you may be missing `ebtables`, `ethtool` or a similar executable on your Linux machine. You can install them with the following commands: 
+Then you may be missing `ebtables`, `ethtool` or a similar executable on your node. You can install them with the following commands:
 
 - For ubuntu/debian users, run `apt install ebtables ethtool`.
```
Comment (neolit123) [inline content/en/docs/setup/independent/troubleshooting-kubeadm.md]: `ubuntu/debian` -> `Ubuntu/Debian`.

**Comment by timothysc** [comment]
/hold 

We need to update the examples around changes from v1alpha1 to v1alpha2, this will likely be a bulk of inbound issues.

File: content/en/docs/setup/independent/troubleshooting-kubeadm.md (hunk 3 above)
Comment (detiber) [reply content/en/docs/setup/independent/troubleshooting-kubeadm.md]: This is the error output from the kubelet when the cgroup driver does not match.

Comment (neolit123) [reply content/en/docs/setup/independent/troubleshooting-kubeadm.md]: oh...

**Comment by timothysc** [comment]
@detiber we can proceed with this and follow up, or add the content now, your choice.

**Comment by detiber** [comment]
@timothysc lets proceed with this for now and we can follow up later. I think the upgrade issues are more pressing atm.

**Pull request merged**